hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/OzoneAcl.java

```diff
@@ -58,17 +59,12 @@ public class OzoneAcl {
   private final AclScope aclScope;
   private static final List<ACLType> EMPTY_LIST = new ArrayList<>(0);
 
-  // TODO use varargs constructor
-  public OzoneAcl(ACLIdentityType type, String name, ACLType acl, AclScope scope) {
-    this(type, name, scope, bitSetOf(acl));
-  }
-
   public OzoneAcl(ACLIdentityType type, String name, AclScope scope, ACLType... acls) {
     this(type, name, scope, bitSetOf(acls));
   }
 
-  public OzoneAcl(ACLIdentityType type, String name, BitSet acls, AclScope scope) {
-    this(type, name, scope, validateAndCopy(acls));
+  public OzoneAcl(ACLIdentityType type, String name, AclScope scope, EnumSet<ACLType> acls) {
+    this(type, name, scope, bitSetOf(acls.toArray(new ACLType[0])));
```

Comment:
The reason of using `toArray(newACLType[0])` is to enforce the runtime type?

Reply:
That's how [`toArray(T[])`](https://docs.oracle.com/javase/8/docs/api/java/util/Collection.html#toArray-T:A-) knows the return type.  Otherwise [`toArray()`](https://docs.oracle.com/javase/8/docs/api/java/util/Collection.html#toArray--) can only return `Object[]`.